### PR TITLE
(feature): Bn254 Curve Implementation

### PR DESCRIPTION
**Pull request by protocolwhisper**
# Bn254 Curve
- [x] Add the finite field used by the curve, using MontgomeryBackend
- [x] Add the curve
## Description

This partially close  https://github.com/lambdaclass/lambdaworks/issues/548

## Type of change

Please delete options that are not relevant.

- [x] New feature
## Checklist
- [x] Linked to Github Issue
- [x] Unit test added

### CODE REVIEW

File: math/src/elliptic_curve/short_weierstrass/curves/bn_254/curve.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+pub struct BN254Curve;
+
+impl IsEllipticCurve for BN254Curve {
+    type BaseField = BN254PrimeField;
+    type PointRepresentation = ShortWeierstrassProjectivePoint<Self>;
+
+    //Affine Coordinates
+    fn generator() -> Self::PointRepresentation {
+        Self::PointRepresentation::new([
+            // I don't believe in this ut i gotta represent this in the other file
```
Comment (MauroToscano) [inline math/src/elliptic_curve/short_weierstrass/curves/bn_254/curve.rs:22]: What's this comment ?

File: math/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "rust-analyzer.linkedProjects": ["./Cargo.toml"]
+}
```
Comment (MauroToscano) [inline math/.vscode/settings.json:3]: This file shouldn't be pushed

**Review by MauroToscano — CHANGES_REQUESTED** [review]

File: math/.vscode/settings.json (hunk 2 above)
Comment (protocolwhisper) [reply math/.vscode/settings.json]: My bad

**Comment by protocolwhisper** [comment]
@MauroToscano I'm closing this PR cause this is being solved in this team PR: https://github.com/lambdaclass/lambdaworks/pull/646

**Pull request closed without merging**